packages/bitcore-wallet-service/src/lib/fiatrateservice.ts

```diff
@@ -168,6 +153,56 @@ export class FiatRateService {
   getRates(opts, cb) {
     $.shouldBeFunction(cb, 'Failed state: type error (cb not a function) at <getRates()>');
 
+    opts = opts || {};
+
+    const now = Date.now();
+    const ts = opts.ts ? opts.ts : now;
+    let fiatFiltered = [];
+    let rates = [];
+
+    if (opts.code) {
+      fiatFiltered = _.filter(Defaults.FIAT_CURRENCIES, ['code', opts.code]);
+      if (!fiatFiltered.length) return cb(opts.code + ' is not supported');
+    }
+    const currencies: { code: string; name: string }[] = fiatFiltered.length ? fiatFiltered : Defaults.FIAT_CURRENCIES;
+
+    async.map(
+      _.values(Constants.COINS),
+      (coin, cb) => {
+        rates[coin] = [];
+        async.map(
+          currencies,
+          (currency, cb) => {
+            this.storage.fetchFiatRate(coin, currency.code, ts, (err, rate) => {
+              if (err) return cb(err);
+              if (rate && ts - rate.ts > Defaults.FIAT_RATE_MAX_LOOK_BACK_TIME * 60 * 1000) rate = null;
+              return cb(null, {
+                ts: +ts,
+                rate: rate ? rate.value : undefined,
+                fetchedOn: rate ? rate.ts : undefined,
+                code: currency.code,
+                name: currency.name
+              });
+            });
+          },
+          (err, res: any) => {
+            if (err) return cb(err);
+            var obj = {};
+            obj[coin] = res;
+            return cb(null, obj);
+          }
+        );
+      },
+      (err, res: any) => {
+        if (err) return cb(err);
+        return cb(null, Object.assign({}, ...res));
+      }
+    );
+  }
+
+  getRatesByCoin(opts, cb) {
+    $.shouldBeFunction(cb, 'Failed state: type error (cb not a function) at <getRatesByCoin()>');
```

Comment:
maybe add:

    if (!checkRequired(opts, ['coin'], cb)) return;
(so, coin is required always).

Reply:
this check is done in the previous step. Check server.ts